travis/travis_weblate.py

```diff
@@ -1,166 +1,216 @@
+            'title': '[REF] i18n: Conflict on the daily cron',
+            'head': '%s:%s' % (self.repo_slug.split('/')[0], branch_name),
+            'base': component['branch'],
+            'body': message
+        })
+        print(yellow("The pull request register is: %s" % pull['html_url']))
+
+    def update(self):
+        self._check()
+        self.wl_api.load_project(self.repo_slug, self.branch)
+        self.wl_api.pull()
+        if not self.wl_api.components:
+            print(yellow("No component found for %s" % self.repo_slug))
+        for component in self.wl_api.components:
+            if component['vcs'] == 'git':
+                first_commit = False
+                name = '%s-weblate' % component['branch']
+                remote = (self.wl_api.host.replace('api', 'git') + '/' +
+                          self.wl_api.project['slug'] + '/' +
+                          component['slug'])
+                self.wl_api.component_unlock(component)
+                self.wl_api.component_lock(component)
+                self.wl_api.component_repository(component, 'pull')
+                self._git.run(["remote", "remove", name])
+                self._git.run(["remote", "add", name, remote])
+                self._git.run(["fetch", "--all", "--depth=1"])
+                self._git.run(["reset", "--hard"])
+                self._git.run(["checkout", "-b", component['branch'],
+                               "origin/%s" % component['branch']])
+                self._git.run(["reset", "--hard",
+                               "origin/%s" % component['branch']])
+                self._git.run(["merge", "--squash",
+                               "%s/%s" % (name, component['branch'])])
+                if self._has_conflict(component):
+                    continue
+                if self._add_odoo_po_files(component):
+                    first_commit = True
+                    self._git.run(["commit", "--no-verify",
+                                   "--author='Weblate bot <weblate@bot>'",
+                                   "-m", "[REF] i18n: Updating translation"
+                                   "terms from weblate [ci skip]"])
+                self._generate_odoo_po_files()
+                if self._has_conflict(component):
+                    continue
+                self._add_odoo_po_files(component)
+                if first_commit:
+                    self._git.run(["commit", "--no-verify", "--amend",
+                                   "--no-edit"])
+                else:
+                    self._git.run(["commit", "--no-verify",
+                                   "--author='Weblate bot <weblate@bot>'",
+                                   "-m", "[REF] i18n: Updating translation"
+                                   "terms from weblate [ci skip]"])
+                self._git.run(["push", "origin", component['branch']])
+                self.wl_api.component_repository(component, 'reset')
```

Comment:
I made the method `_push_git_repository` which checks if there is something to commit and then restart the repository